brave/src/main/java/brave/internal/handler/MutableSpanConverter.java

```diff
@@ -65,38 +76,30 @@ void convert(MutableSpan span, Span.Builder result) {
         .build());
     }
 
-    String errorTag = span.tag("error");
-    if (errorTag == null && span.error() != null) {
-      errorParser.error(span.error(), span);
-    }
-
     span.forEachTag(Consumer.INSTANCE, result);
     span.forEachAnnotation(Consumer.INSTANCE, result);
     if (span.shared()) result.shared(true);
+    if (span.debug()) result.debug(true);
   }
 
   // avoid re-allocating an endpoint when we have the same data
-  void addLocalEndpoint(String serviceName, @Nullable String ip, int port, Span.Builder span) {
-    if (serviceName == null) serviceName = localServiceName;
-    if (ip == null) ip = localIp;
-    if (port <= 0) port = localPort;
-    if (localServiceName.equals(serviceName)
-      && (localIp == null ? ip == null : localIp.equals(ip))
-      && localPort == port) {
-      span.localEndpoint(localEndpoint);
+  void addLocalEndpoint(@Nullable String serviceName, @Nullable String ip, int port, Builder span) {
+    if (serviceName != null) serviceName = serviceName.toLowerCase(Locale.ROOT);
+    if (hashEndpointParameters(serviceName, ip, port) == defaultEndpointHashCode) {
```

Comment:
Think you still need to compare the values in case of hash collision

Reply:
fair enough :)

Reply:
I didn't think about this. luckily somewhere I have a string that has a different value with same hashcode.. /me digs

Reply:
thanks for the tip on gitter https://dzone.com/articles/what-is-wrong-with-hashcode-in-javalangstring